### PR TITLE
Feature/cu p5b6hz HUD Rework

### PR DESCRIPTION
**Pull request by GBMiro**
- Changed skill available effect with new textures
- Added new effects when switching characters
- Changed health bars with new textures
- Other new textures
- Refactor

Test scene: Level1_NEW_HUD_Final.scene

PD: The HUD won't work in Level1.scene due to changes in the HUD hierarchy. It needs to be set up. In the Level1_NEW_HUD_Final.scene dialogs don't work because are not properly set (scene is old) but in Level1.scene they still work.

### CODE REVIEW

**Comment by jcarlos0305** [comment]
Task linked: [CU-p5b6hz UI HUD Redesign](https://app.clickup.com/t/p5b6hz)

**Review by Mefiso — COMMENTED** [review]
Should the dialogues be working on the scene Level1_NEW_HUD_Final ? Because in mine they don't.

**Comment by GBMiro** [comment]
> Should the dialogues be working on the scene Level1_NEW_HUD_Final ? Because in mine they don't.

They should not work because they are not set up. In Level1.scene they must work

File: Penteract/Assets/Scripts/HUDManager.cpp [hunk 1]
```diff
@@ -177,17 +189,24 @@ void HUDManager::Update() {
...
-	cooldowns[static_cast<int>(Cooldowns::FANG_SKILL_3)] = fangUltimateRemainingNormalizedValue == 0 ? fangCooldown3 : 1.0f;
+	cooldowns[static_cast<int>(Cooldowns::FANG_SKILL_3)] = fangUltimateRemainingNormalizedValue == 0 ? fangCooldown3 : 0.99f;
 	cooldowns[static_cast<int>(Cooldowns::ONIMARU_SKILL_1)] = onimaruCooldown1;
 	cooldowns[static_cast<int>(Cooldowns::ONIMARU_SKILL_2)] = onimaruCooldown2;
-	cooldowns[static_cast<int>(Cooldowns::ONIMARU_SKILL_3)] = oniUltimateRemainingNormalizedValue == 0 ? onimaruCooldown3 : 1.0f;
+	cooldowns[static_cast<int>(Cooldowns::ONIMARU_SKILL_3)] = oniUltimateRemainingNormalizedValue == 0 ? onimaruCooldown3 : 0.99f;
 	cooldowns[static_cast<int>(Cooldowns::SWITCH_SKILL)] = switchCooldown;
 
+
+	//std::string message = std::to_string(fangCooldown1) + " " + std::to_string(fangCooldown2) + " " + std::to_string(fangCooldown3) + " " +
```
Comment (magalenyo) [inline Penteract/Assets/Scripts/HUDManager.cpp:205]: Dead code should be removed

File: Penteract/Assets/Scripts/HUDManager.cpp [hunk 2]
```diff
@@ -177,17 +189,24 @@ void HUDManager::Update() {
...
 	ManageSwitch();
 	if (playingLostHealthFeedback) PlayLostHealthFeedback();
+	if (playingHitEffect) PlayHitEffect();
 }
 
 void HUDManager::UpdateCooldowns(float onimaruCooldown1, float onimaruCooldown2, float onimaruCooldown3, float fangCooldown1, float fangCooldown2, float fangCooldown3, float switchCooldown, float fangUltimateRemainingNormalizedValue, float oniUltimateRemainingNormalizedValue) {
 	cooldowns[static_cast<int>(Cooldowns::FANG_SKILL_1)] = fangCooldown1;
 	cooldowns[static_cast<int>(Cooldowns::FANG_SKILL_2)] = fangCooldown2;
-	cooldowns[static_cast<int>(Cooldowns::FANG_SKILL_3)] = fangUltimateRemainingNormalizedValue == 0 ? fangCooldown3 : 1.0f;
+	cooldowns[static_cast<int>(Cooldowns::FANG_SKILL_3)] = fangUltimateRemainingNormalizedValue == 0 ? fangCooldown3 : 0.99f;
```
Comment (magalenyo) [inline Penteract/Assets/Scripts/HUDManager.cpp:198]: why is this a .99f?

Comment (GBMiro) [reply Penteract/Assets/Scripts/HUDManager.cpp:198]: To solved an effect triggering bug with the abilities

**Review by magalenyo — DISMISSED** [review]

**Review by magalenyo — APPROVED** [review]

**Review by Mefiso — DISMISSED** [review]

**Review by jcarlos0305 — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]

**Review by magalenyo — APPROVED** [review]

File: Penteract/Assets/Scripts/HUDManager.cpp [hunk 3]
```diff
@@ -279,9 +284,21 @@ void HUDManager::StartCharacterSwitch() {
...
 		}
 	}
 	if (playingLostHealthFeedback) StopLostHealthFeedback();
+		
+	// Check if the new character needs the health warning
+
+	float health = fangObj->IsActive() ? onimaruPreviousHealth : fangPreviousHealth;
+	float maxHealth = fangObj->IsActive() ? playerController->GetOnimaruMaxHealth() : playerController->GetFangMaxHealth();
+
+	if (health > maxHealth * (criticalHealthPercentage / 100.f)) {
```
Comment (Mefiso) [inline Penteract/Assets/Scripts/HUDManager.cpp:298]: What if we try to use 0.whatever instead of whatever/100.f

Comment (devildrake) [reply Penteract/Assets/Scripts/HUDManager.cpp:298]: Donezel Washington

**Review by Mefiso — APPROVED** [review]

**Pull request merged**